File: payroll-api/src/main/java/com/company/payroll/mapper/PromotionMapper.java
package com.company.payroll.mapper;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.company.payroll.model.Promotion;

@Repository
public interface PromotionMapper {
    int deleteByPrimaryKey(Integer pId);

    int insert(Promotion row);

    int insertSelective(Promotion row);
    
    List<Promotion> selectByEId(Integer eId);
    
    Promotion selectByPrimaryKey(Integer pId);

    List<Promotion> selectList();

    int updateByPrimaryKey(Promotion row);

    int updateByPrimaryKeySelective(Promotion row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/PromotionMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.Promotion
import org.springframework.stereotype.Repository

@Repository
interface PromotionMapper {
  fun deleteByPrimaryKey(pId: Int): Int
  fun insert(row: Promotion): Int
  fun insertSelective(row: Promotion): Int
  fun selectByEId(staffId: Int): List<Promotion>?
  fun selectByPrimaryKey(pId: Int): Promotion?
  fun selectList(): List<Promotion>?
  fun updateByPrimaryKey(row: Promotion): Int
  fun updateByPrimaryKeySelective(row: Promotion): Int
}